src/components/picker/types.tsx

```diff
@@ -147,49 +122,111 @@ export type PickerBaseProps = Omit<NewTextFieldProps, 'value' | 'onChange'> & {
    * Render custom search input (only when passing showSearch)
    */
   renderCustomSearch?: (props: PickerItemsListProps) => React.ReactElement;
+};
+
+type PickerListProps = PickerSearchProps & {
   /**
-   * Render a custom header for Picker's dialog
-   */
-  renderCustomDialogHeader?: (callbacks: {onDone?: () => void, onCancel?: ()=> void}) => React.ReactElement;
-  /**
-   * Use wheel picker instead of a list picker
+   * Render a custom header for Picker's Overlay
    */
-  useWheelPicker?: boolean;
+  renderHeader?: (callbacks: {onDone?: () => void; onCancel?: () => void}) => React.ReactElement;
   /**
-   * Pass props to the list component that wraps the picker options (allows to control FlatList behavior)
+   * Pass props to the list component that wraps the picker items (allows to control FlatList behavior)
    */
   listProps?: Partial<FlatListProps<any>>;
   /**
-   * Pass props to the picker modal
-   */
-  pickerModalProps?: object;
-  /**
-   * Custom container style
-   */
-  containerStyle?: StyleProp<ViewStyle>;
-  /**
-   * @deprecated
-   * Callback for modal onShow event
+   * The picker modal top bar props
    */
-  onShow?: () => void;
+  topBarProps?: ModalTopBarProps;
   /**
    * Add safe area in the Picker modal view
    */
   useSafeArea?: boolean;
+};
+
+type PickerExpandableOverlayProps = {
   /**
-   * Data source for Picker
+   * Custom picker overlay props
    */
-  items?: PickerItemProps[];
+  customPickerProps?: ExpandableOverlayProps;
   /**
-   * Component test id
+   * Render custom picker overlay (e.g ({visible, children, toggleModal}) => {...})
    */
-  testID?: string;
+  renderOverlay?: (modalProps: RenderCustomModalProps) => React.ReactElement;
   /**
-   * @deprecated
+   * Add blur effect to picker modal (iOS only)
    */
-  children?: ReactNode | undefined;
+  enableModalBlur?: boolean;
 };
 
+export type PickerBaseProps = Omit<TextFieldProps, 'value' | 'onChange'> &
+  PickerPropsDeprecation &
+  PickerExpandableOverlayProps &
+  PickerListProps & {
+    /* ...TextField.propTypes, */
+    /**
+     * Use dialog instead of modal picker
+     */
+    useDialog?: boolean;
+    /**
+     * Use wheel picker instead of a list picker
+     */
+    useWheelPicker?: boolean;
+    /**
+     * Picker current value in the shape of {value: ..., label: ...}, for custom shape use 'getItemValue' prop
```

Comment:
getItemValue is deprecated...

Reply:
Also the shape isn't right.
The type is string | number or array of that.
```suggestion
     * Picker value
```